syft/pkg/cataloger/apkdb/parse_apk_db.go

```diff
@@ -20,11 +22,41 @@ import (
 // integrity check
 var _ generic.Parser = parseApkDB
 
+var (
+	repoRegex         = regexp.MustCompile(`^https://.*\.alpinelinux\.org/alpine/v([^\/]+)/([a-zA-Z0-9_]+)$`)
+	newlineSplitRegex = regexp.MustCompile(`[\r \s\n]+`)
+)
+
+func init() {
+	repoRegex.Longest()
+	newlineSplitRegex.Longest()
+}
```

Comment:
I think this could be simplified to use a multiline modifier (`(?m)`) in the original regex, e.g:

```go
repoRegex = regexp.MustCompile(`(?m)^https://.*\.alpinelinux\.org/alpine/v([^\/]+)/([a-zA-Z0-9_]+)$`)
```

... and since it reads the entire line (the `^` and `$`), it probably doesn't need the `.Longest()` modifier.

see: https://regex101.com/r/goXXNl/1

Reply:
_That_ is a good approach. Pity that I deleted the play.golang where I was testing all of it. Oh well.